### DISCUSSION TITLE
Compatibility with Wireplumber's bluetooth.autoswitch-to-headset-profile

### DISCUSSION THREAD

**Discussion by rivenirvana** [wwmm/easyeffects#3488] [Q&A]
I'm trying to use Bluetooth earbuds on my Linux setup and it took me a while to find out that easyeffects was the one preventing wireplumber's `bluetooth.autoswitch-to-headset-profile` (switching between HSP/HFP and A2DP whenever an input stream is detected or closed) setting to work properly. I had easyeffects on autostart and quitting it made wireplumber work properly. I'm not using a very custom setup on easyeffects, but resetting my preferences does not make it work, as well.

May I have some guidance on if it is possible to have compatibility with wireplumber in this regard? I'd like to be able to keep using this application for my minimal needs, any pointers would be appreciated.

Easyeffects v7.1.9 (RPM)
Wireplumber v0.5.5
Pipewire v1.2.6 (no pipewire-media-session)

**Comment by wwmm** [comment]
That is odd. I do not have problems manually switching the profile of my bluetooth headphone while EasyEffects is running. I wonder how the automatic switch done by wireplumber works to be having issues with this.

Run the command `pw-dot` while EasyEffects is running the the problems happens. The output file can be viewed with the command `xdot` and will show how filters and devices are linked.

>switching between HSP/HFP and A2DP whenever an input stream is detected or closed

My guess for now is that there is a limitation in wireplumber. Filters and streams are different objects in PipeWire. If Wireplumber relies on streams being linked to the bluetooth microphone to do the automatic switching this will never work with EasyEffects because our filters are not streams. What also means that the links to the microphone are handled in a totally different way.

**Comment by rivenirvana** [comment]
Yes, I also don't have problems manually switching the profiles. I'll keep looking into it in the meantime.

**Comment by rivenirvana** [comment]
Well, this is a bit embarrassing, but I guess my issue is fixed now. Reset easyeffects' settings again, made sure to remove any stray user configs for pipewire, pulse, also removed wireplumber states in `.local/state`, then gave it a reboot. Properly autoswitches Bluetooth profiles now for applications that open an input stream.

For what it's worth though, enabling `Process all input streams` in Easyeffects' preferences (disabled by default) does interfere with Wireplumber's autoswitching, but I'm guessing this is expected behavior? In any case, will be closing this now.

**Reply by wwmm** [reply]
> For what it's worth though, enabling `Process all input streams` in Easyeffects' preferences (disabled by default) does interfere with Wireplumber's autoswitching, but I'm guessing this is expected behavior?

When this is enabled EasyEffects asks wireplumber to move recording streams to our virtual sources. In theory this shoudn't mess with bluetooth hardware profiles. It will depend on how wireplumber is managing the automatic profile switching.
